crates/ruma-federation-api/src/transactions/edu.rs

```diff
@@ -341,6 +342,43 @@ mod test {
         );
     }
 
+    #[test]
+    fn minimal_device_list_update_edu() {
+        let json = json!({
+                "content": {
+                    "device_id": "QBUAZIFURK",
+                    "stream_id": 6,
+                    "user_id": "@john:example.com"
+                },
+            "edu_type": "m.device_list_update"
+        });
+
+        let edu = serde_json::from_value::<Edu>(json).unwrap();
+        assert_matches!(
+            &edu,
+            Edu::DeviceListUpdate(DeviceListUpdateContent {
+                user_id,
+                device_id,
+                device_display_name,
+                stream_id,
+                prev_id,
+                deleted,
+                keys
+            }) if user_id == &user_id!("@john:example.com")
+                && device_id == &device_id!("QBUAZIFURK")
+                && device_display_name.is_none()
+                && stream_id == &uint!(6)
```

Comment:
Shouldn't be necessary to convert things to ID types just for comparison. And I think dereferencing one side of a comparison is more common than adding a reference on the other.

```suggestion
            }) if user_id == "@john:example.com"
                && device_id == "QBUAZIFURK"
                && device_display_name.is_none()
                && *stream_id == uint!(6)
```

Reply:
I've changed it in this test and the others in this file, for consistency